src/url-params.ts

```diff
@@ -111,6 +111,15 @@ export class AnalyticsUrlParams {
 		this.client = client;
 	}
 
+	/**
+	 *
+	 * @param client
+	 * Sets the class client value to whatever client object is passed on the argument.
+	 */
+	declareClient(client: Client) {
```

Comment:
I think the idea is decent and it is a needed change.
Thinking further, do we have any other naming suggestions? I'm trying to imagine how confusing it is to have `setClient` and `declareClient`.

Reply:
Right, I have the same feeling that is not the best name, but then again, what we are doing is actually setting the client value on the AnalyticsUrlParams class, but setClient is already taken for a much bigger purpose, and I thought that event having something like setClassClient , including both 'set' and 'client' would be more confusing than 'declareClient'. But Im open to any suggestion!